src/server/passive_tran_server.hpp

```diff
@@ -27,9 +28,16 @@ class passive_tran_server : public tran_server
     passive_tran_server () : tran_server (cubcomm::server_server::CONNECT_PASSIVE_TRAN_TO_PAGE_SERVER)
     {
     }
+    ~passive_tran_server () override;
 
   public:
     void send_and_receive_log_boot_info (THREAD_ENTRY *thread_p);
+    void start_log_replicator (const log_lsa &start_lsa);
+
+  public:
+    /* read replicator's current progress */
+    log_lsa get_replicator_lsa () const;
+    void finish_replication_during_shutdown (cubthread::entry &thread_entry);
```

Comment:
No reason for having separated public sections here:
```suggestion
  public:
    void send_and_receive_log_boot_info (THREAD_ENTRY *thread_p);
    void start_log_replicator (const log_lsa &start_lsa);
    /* read replicator's current progress */
    log_lsa get_replicator_lsa () const;
    void finish_replication_during_shutdown (cubthread::entry &thread_entry);
```